pkg/aws/aws_client/role.go

```diff
@@ -292,3 +305,52 @@ func completeRolePolicyDocument(statement map[string]interface{}) (string, error
 	assumeRolePolicyDocument, err := json.Marshal(rolePolicyDocument)
 	return string(assumeRolePolicyDocument), err
 }
+
+func (client *AWSClient) AttachPolicy(roleName string, policyArn string) error {
+	policyAttach := iam.AttachRolePolicyInput{
+		PolicyArn: &policyArn,
+		RoleName:  &roleName,
+	}
+	_, err := client.IamClient.AttachRolePolicy(context.TODO(), &policyAttach)
+	if err != nil {
+		return err
+	}
+	timeout := 2
+	start := 0
+
+	for start < timeout {
```

Comment:
@OAharoni-RedHat Personally, I'd either:

1. Let the client do the retry themselves; or
2. Allow the client to provide via method parameters the configuration for the retry

At the minute this method is not really unit-testable without significantly extending the duration of the test suite. There is also no logging (or indication) to clients that calling this is potentially a very long wait.

I'd _personally_ lean towards `1` as it just makes use of methods you have implemented here e.g:

```
err := client.AttachPolicy("foo", "bar")
if err != nil {
 // check to be absolutely sure it's attached
 attached, err := client.PolicyAttachedToRole("foo", "bar")
 if !attached {
  // some configurable retry logic here
```

Alternatively you could define a "retry" struct that gets passed in as a method parameter that makes it easy for the client to control behaviour here e.g:

```
type RetryConfig struct {
 maxRetries int
 waitDuration time.Duration
}
```

And then your method looks _something_ this:

```
func (client *AWSClient) AttachPolicy(roleName string, policyArn string, retryConfig RetryConfig) error {
	policyAttach := iam.AttachRolePolicyInput{
		PolicyArn: &policyArn,
		RoleName:  &roleName,
	}
	_, err := client.IamClient.AttachRolePolicy(context.TODO(), &policyAttach)
	if err != nil {
		return err
	}

       if attached, _ := client.PolicyAttachedToRole(roleName, policyArn); attached {
			return nil
      }

     // Retry based on fields of RetryConfig
}